source/extensions/listener_managers/listener_manager/filter_chain_manager_impl.cc

```diff
@@ -849,5 +913,30 @@ Configuration::FilterChainFactoryContextPtr FilterChainManagerImpl::createFilter
   return std::make_unique<PerFilterChainFactoryContextImpl>(parent_context_, init_manager_);
 }
 
+FcdsApiPtr FilterChainManagerImpl::createFcdsApi(const envoy::config::listener::v3::ListenerFcds& fcds_config,
+		                                 FilterChainFactoryBuilder* fc_builder,
+						                         ListenerManager* listener_mgr,
+						                         std::string listener_name) {
+	listener_manager_ = listener_mgr;
+	listener_name_ = listener_name;
+	return std::make_shared<FcdsApi>(fcds_config, parent_context_, init_manager_, *this, fc_builder);
+}
+
+void FilterChainManagerImpl::addFcToDrainingList(Network::DrainableFilterChainSharedPtr fc) {
+	this->draining_fc_list_.push_back(fc);
+}
+
+void FilterChainManagerImpl::removeFcFromFcdsDrainingList(Network::DrainableFilterChainSharedPtr fc) {
```

Comment:
Where is this being used?

Reply:
In my previous diff, this method was used in findExistingFilterChain; when we find that the filter chain added in the drainingfc_list initially, is actually unchanged, and thus its removed from the draiing_fc_list. That code is not needed now, as we are doing this check now much earlier, before even adding an FC in the draining_fc_list, we are ensuring its indeed modified or removed.

Hence, this is a dead code, am removing it.